addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanner.java

```diff
@@ -0,0 +1,144 @@
+
+    static {
+        CONTENT_ATTRIBUTES.put(SCRIPT, "src");
+        CONTENT_ATTRIBUTES.put(LINK, "href");
+    }
+
+    // TODO Replace "rules.domains.trusted" with RuleConfigParam.RULE_DOMAINS_TRUSTED once
+    // available.
+    static final String TRUSTED_DOMAINS_PROPERTY = "rules.domains.trusted";
+
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        Collection<String> trustedDomains =
+                Stream.of(getConfig().getString(TRUSTED_DOMAINS_PROPERTY, "").split(","))
+                        .filter(s -> !s.isEmpty())
+                        .collect(Collectors.toList());
+
+        List<Element> sourceElements = source.getAllElements();
+        sourceElements.stream()
+                .filter(element -> SUPPORTED_ELEMENTS.contains(element.getName()))
+                .filter(excludeSafeResourceFrom(trustedDomains))
+                .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
+                .forEach(
+                        element -> {
+                            Alert alert =
+                                    new Alert(
+                                            getPluginId(),
+                                            Alert.RISK_MEDIUM,
+                                            Alert.CONFIDENCE_HIGH,
+                                            getName());
+
+                            alert.setDetail(
+                                    getString("desc"),
+                                    msg.getRequestHeader().getURI().toString(),
+                                    "",
+                                    "",
+                                    "",
+                                    getString("soln"),
+                                    getString("refs"),
+                                    element.toString(),
+                                    16, // CWE CATEGORY: Configuration
+                                    15, // Application Misconfiguration
+                                    msg);
+                            parent.raiseAlert(id, alert);
+                        });
+    }
+
+    private Predicate<Element> excludeSafeResourceFrom(Collection<String> trustedDomains) {
+        return element -> {
+            String domain = element.getAttributeValue(CONTENT_ATTRIBUTES.get(element.getName()));
+            return trustedDomains.stream().noneMatch(domain::matches);
+        };
+    }
+
+    private static Predicate<Element> unsafeSubResource(String hostname) {
+        return element ->
+                element.getAttributeValue("integrity") == null
+                        && !element.getAttributeValue(CONTENT_ATTRIBUTES.get(element.getName()))
+                                .matches("^https?://[^/]*" + hostname + "/.*");
```

Comment:
- It would be better if the pattern was compiled once.
 - The scheme should be optional as well.

Reply:
After some tests, I think I'm not sure of what should be done.
Trusted domains include scheme, domains and ports. So, I think it should be renamed as "trusted.urls", to express that it is not only the domains.

Also, should we apply the same rule as the "Same Origin Policy"? Meaning everything with a different domain/scheme/port/... should be considered as unsafe?
I think I should also do some tests with more complex contents (like `http://domain////`)

Reply:
That would be better but not sure worth renaming it at this point, the description of the config is clear that it has URL regexes and renaming it would "break" existing rules using it.

That's a good point, I think the other rules only check (mainly) the domain/port. Probably something to revise in a follow up PR (like the refactoring)?